Katello/katello
# Fixes #31928 - Add auth_url and token fields for ansible collections

1. This adds the fields auth_url and token fields to repository and sends them to pulpcore api to set these on collection remote.

**To test:**

- Create an ansible collection Repository.
- Set some random auth URL and token (We are not enforcing any validations from our end)
- Save and check the remote on pulp3. These fields should be set.
 `curl --cert /etc/pki/katello/certs/pulp-client.crt --key /etc/pki/katello/private/pulp-client.key https://`hostname`/pulp/api/v3/remotes/ansible/collection/ | python -m json.tool`

________________
2. This also handles https://projects.theforeman.org/issues/31926:

**To test:**

- On master try creating an ansible collection repository with upstream URL as `https://galaxy.ansible.com/api/v2/collections/testing/k8s_demo_collection/ ` with the trailing slash.
You'll see an error "Invalid URL https://galaxy.ansible.com/api/v2/collections/testing/k8s_demo_collection. Ensure the URL ends '/'."
- On this branch you should be able to create a repository with `https://galaxy.ansible.com/api/v2/collections/testing/k8s_demo_collection/`
